### ISSUE TITLE
Bool vector type mismatch between scalar and SIMD implementations

### ISSUE THREAD

**Issue by Radbuglet** [bitshifter/glam-rs#364]
The type of the boolean vectors used by scalar and SIMD implementations of the same vector type seems to be different.

Consider, for example, `Vec3A`. The SIMD implementation of the vector exposes [this function signature](https://github.com/bitshifter/glam-rs/blob/a7f494fc0d95f888d689ba3110a6f2d14e48e372/src/f32/sse2/vec3a.rs#L92):

```rust
pub fn select(mask: BVec3A, if_true: Self, if_false: Self) -> Self {
    // ...
}
```

whereas the scalar implementation of the vector exposes [this different function signature](https://github.com/bitshifter/glam-rs/blob/a7f494fc0d95f888d689ba3110a6f2d14e48e372/src/f32/scalar/vec3a.rs#L88):

```rust
pub fn select(mask: BVec3, if_true: Self, if_false: Self) -> Self {
    // ...
}
```

This seems to be a regression from `0.22.0`, which:

> Added `u32` implementation of `BVec3A` and `BVec4` when SIMD is not available. These are used instead of aliasing to the `bool` implementations.

Presumably, this problem wasn't noticed before because, on scalar platforms, `BVec3` and `BVec3A` were interchangeable.

**Comment by bitshifter** [comment]
Hmm, that is indeed wrong.

**Comment by bitshifter** [comment]
I've fixed this in main, I'll leave this open until I've made a new release.

**Comment by bitshifter** [comment]
Fix in 0.23.0 release.
